tests/RobotFramework/tests/cumulocity/service_monitoring/service_monitoring.robot

```diff
@@ -174,30 +172,74 @@ Check if a service is down
 
 Check if a service using configured service type
     [Arguments]    ${service_name}
-    Execute Command    tedge config set service.type thinedge
+
     Custom Test Setup
+
+    ${previous_service_type}=     Execute Command    tedge config get service.type    strip=True
+    Execute Command               tedge config set service.type thinedge
+
     ThinEdgeIO.Restart Service    ${service_name}
-    Device Should Exist                      ${DEVICE_SN}:device:main:service:${service_name}    show_info=False
+    Device Should Exist           ${DEVICE_SN}:device:main:service:${service_name}    show_info=False
+
+    # If service was previously registered, old serviceType should be maintained
```

Comment:
I'm doubtful with this requirement. I think keeping track of what the previously registered type is will be overly complex to maintain and reason with...but I think it would be a good opportunity creating some sequence diagrams showing the interactions between the registration messages and the `/status/health` telemetry data points.

So maybe once we have that, we can think of how to simplify this.

Reply:
The explicit registration was added and the change to the test was reverted.